### PR TITLE
Allow adding more table indexing configuration for Pinot tables.

### PR DESCRIPTION
**Pull request by buchi-busireddy**
Currently, we only allow the inverted indices to be added to Pinot table but
we need to be able to add other indexes like bloom filters, range indexes
to optimize the performance of Pinot queries. This PR let's that config to
be tunable easily.

I'm also rethinking if we should have this indirection in Pinot table config
or should we directly provide the JSON that's needed but that's for future PRs.

Also upgraded the pinot-tools to 0.5.0 version in this PR.

### CODE REVIEW

File: view-creator-framework/src/test/java/org/hypertrace/core/viewcreator/pinot/PinotUtilsTest.java [hunk 1]
```diff
@@ -59,54 +60,57 @@ public void testCreatePinotTableForView() {
...
-    Assertions.assertEquals(
-        tableConfig.getIndexingConfig().getStreamConfigs().get("stream.kafka.decoder.prop.schema.registry.rest.url"),
-        "http://localhost:8081");
-    Assertions.assertEquals(tableConfig.getIndexingConfig().getStreamConfigs()
-        .get("realtime.segment.flush.threshold.time"), "3600000");
-    Assertions.assertEquals(tableConfig.getIndexingConfig().getStreamConfigs()
-        .get("realtime.segment.flush.threshold.size"), "500000");
-    Assertions.assertEquals(tableConfig.getIndexingConfig().getStreamConfigs()
-        .get("stream.kafka.consumer.prop.auto.offset.reset"), "largest");
+    Assertions.assertEquals("myView1_REALTIME", tableConfig.getTableName());
```
Comment (buchi-busireddy) [inline view-creator-framework/src/test/java/org/hypertrace/core/viewcreator/pinot/PinotUtilsTest.java:66]: The order of `expected` and `actual` arguments to `assertEquals` method was wrong here so I have fixed them all to be correct.

File: view-creator-framework/build.gradle.kts [hunk 2]
```diff
@@ -92,11 +92,17 @@ dependencies {
       because("XML Entity Expansion [High Severity][https://snyk.io/vuln/SNYK-JAVA-ORGGLASSFISHJERSEYMEDIA-595972] in org.glassfish.jersey.media:jersey-media-jaxb@2.28\n" +
               "    introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.pinot:pinot-common@0.3.0 > org.glassfish.jersey.core:jersey-server@2.28 > org.glassfish.jersey.media:jersey-media-jaxb@2.28 and 1 other path(s)\n")
     }
+    implementation("io.grpc:grpc-core@1.31.0") {
```
Comment (tim-mwangi) [inline view-creator-framework/build.gradle.kts:95]: why is this needed?

File: view-creator-framework/src/test/java/org/hypertrace/core/viewcreator/pinot/PinotUtilsTest.java (hunk 1 above)
Comment (tim-mwangi) [reply view-creator-framework/src/test/java/org/hypertrace/core/viewcreator/pinot/PinotUtilsTest.java:66]: I thought I had fixed when I was moving this around.

File: view-creator-framework/build.gradle.kts [hunk 3]
```diff
@@ -92,11 +92,17 @@ dependencies {
...
+      version {
+        because("Information Exposure [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-IOGRPC-571957] in io.grpc:grpc-core@1.30.0")
+        require("1.31.0")
```
Comment (aaron-steinfeld) [inline view-creator-framework/build.gradle.kts:98]: Cleaned up the unused constraints and fixed this - the issue is a bad dependency notation on 95, using `@` rather than `:` to separate the version.

```diff
diff --git a/view-creator-framework/build.gradle.kts b/view-creator-framework/build.gradle.kts
index 4d4a7ee..7576ee3 100644
--- a/view-creator-framework/build.gradle.kts
+++ b/view-creator-framework/build.gradle.kts
@@ -6,15 +6,10 @@ plugins {
   id("org.hypertrace.jacoco-report-plugin")
 }
 
-repositories {
-  // Need this to fetch confluent's kafka-clients dependency
-  maven("http://packages.confluent.io/maven")
-}
-
 sourceSets {
   test {
     java {
-      srcDirs("src/test/java", "build/generated-test-avro-java")
+      srcDirs("build/generated-test-avro-java")
     }
   }
 }
@@ -27,76 +22,32 @@ dependencies {
   implementation("org.hypertrace.core.eventstore:event-store:0.1.1")
   implementation("org.hypertrace.core.serviceframework:platform-service-framework:0.1.13")
   constraints {
-    implementation("com.google.guava:guava:29.0-jre") {
-      because("Deserialization of Untrusted Data [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEGUAVA-32236] in com.google.guava:guava@20.0\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.pinot:pinot-common@0.3.0 > com.google.guava:guava@20.0")
-    }
     implementation("io.netty:netty-all:4.1.50.Final") {
-      because("HTTP Request Smuggling [High Severity][https://snyk.io/vuln/SNYK-JAVA-IONETTY-559515] in io.netty:netty-all@4.1.28.Final\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.pinot:pinot-core@0.3.0 > io.netty:netty-all@4.1.28.Final")
-    }
-    implementation("com.jcraft:jsch:0.1.55") {
-      because("Directory Traversal [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-COMJCRAFT-30302] in com.jcraft:jsch@0.1.42\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.hadoop:hadoop-common@2.7.0 > com.jcraft:jsch@0.1.42")
-    }
-    implementation("commons-codec:commons-codec:1.14") {
-      because("Information Exposure [Low Severity][https://snyk.io/vuln/SNYK-JAVA-COMMONSCODEC-561518] in commons-codec:commons-codec@1.10\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.hadoop:hadoop-common@2.7.0 > commons-codec:commons-codec@1.10")
+      because("HTTP Request Smuggling [High Severity][https://snyk.io/vuln/SNYK-JAVA-IONETTY-559515] in io.netty:netty-all@4.1.28.Final")
     }
     implementation("commons-collections:commons-collections:3.2.2") {
-      because("Deserialization of Untrusted Data [High Severity][https://snyk.io/vuln/SNYK-JAVA-COMMONSCOLLECTIONS-30078] in commons-collections:commons-collections@3.2.1\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.hadoop:hadoop-common@2.7.0 > commons-collections:commons-collections@3.2.1")
-    }
-    implementation("org.apache.hadoop:hadoop-common:3.2.1") {
-      because("Arbitrary File Write via Archive Extraction (Zip Slip) [High Severity][https://snyk.io/vuln/SNYK-JAVA-ORGAPACHEHADOOP-174573] in org.apache.hadoop:hadoop-common@2.7.0\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.hadoop:hadoop-common@2.7.0")
+      because("Deserialization of Untrusted Data [High Severity][https://snyk.io/vuln/SNYK-JAVA-COMMONSCOLLECTIONS-30078] in commons-collections:commons-collections@3.2.1")
     }
     implementation("org.apache.thrift:libthrift:0.13.0") {
-      because("Denial of Service (DoS) [High Severity][https://snyk.io/vuln/SNYK-JAVA-ORGAPACHETHRIFT-474610] in org.apache.thrift:libthrift@0.12.0\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.pinot:pinot-common@0.3.0 > org.apache.thrift:libthrift@0.12.0 and 7 other path(s)")
+      because("Denial of Service (DoS) [High Severity][https://snyk.io/vuln/SNYK-JAVA-ORGAPACHETHRIFT-474610] in org.apache.thrift:libthrift@0.12.0")
     }
     implementation("org.webjars:swagger-ui:3.27.0") {
-      because("Reverse Tabnabbing [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-ORGWEBJARS-449821] in org.webjars:swagger-ui@2.2.2\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.pinot:pinot-common@0.3.0 > org.webjars:swagger-ui@2.2.2")
+      because("Reverse Tabnabbing [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-ORGWEBJARS-449821] in org.webjars:swagger-ui@2.2.2")
     }
     implementation("org.yaml:snakeyaml:1.26") {
-      because("Denial of Service (DoS) [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-ORGYAML-537645] in org.yaml:snakeyaml@1.23\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.yaml:snakeyaml@1.23")
-    }
-    implementation("commons-beanutils:commons-beanutils:1.9.4") {
-      because("Deserialization of Untrusted Data [High Severity][https://snyk.io/vuln/SNYK-JAVA-COMMONSBEANUTILS-460111] in commons-beanutils:commons-beanutils@1.9.3\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.hadoop:hadoop-common@3.2.1 > commons-beanutils:commons-beanutils@1.9.3")
-    }
-    implementation("com.nimbusds:nimbus-jose-jwt:8.19") {
-      because("Improper Check for Unusual or Exceptional Conditions [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-COMNIMBUSDS-536068] in com.nimbusds:nimbus-jose-jwt@4.41.1\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.hadoop:hadoop-common@3.2.1 > org.apache.hadoop:hadoop-auth@3.2.1 > com.nimbusds:nimbus-jose-jwt@4.41.1")
-    }
-    implementation("org.eclipse.jetty:jetty-http:9.4.30.v20200611") {
-      because("Denial of Service (DoS) [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-174011] in org.eclipse.jetty:jetty-http@9.3.24.v20180605\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.hadoop:hadoop-common@3.2.1 > org.eclipse.jetty:jetty-server@9.3.24.v20180605 > org.eclipse.jetty:jetty-http@9.3.24.v20180605")
-    }
-    implementation("org.eclipse.jetty:jetty-server:9.4.30.v20200611") {
-      because("Information Exposure [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-174560] in org.eclipse.jetty:jetty-server@9.3.24.v20180605\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.hadoop:hadoop-common@3.2.1 > org.eclipse.jetty:jetty-server@9.3.24.v20180605")
+      because("Denial of Service (DoS) [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-ORGYAML-537645] in org.yaml:snakeyaml@1.23")
     }
     implementation("org.apache.zookeeper:zookeeper:3.6.1") {
-      because("Access Control Bypass [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-ORGAPACHEZOOKEEPER-174781] in org.apache.zookeeper:zookeeper@3.4.13\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.pinot:pinot-common@0.3.0 > org.apache.zookeeper:zookeeper@3.4.13")
+      because("Access Control Bypass [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-ORGAPACHEZOOKEEPER-174781] in org.apache.zookeeper:zookeeper@3.4.13")
     }
     implementation("io.netty:netty:3.10.6.Final") {
-      because("Information Exposure [High Severity][https://snyk.io/vuln/SNYK-JAVA-IONETTY-30430] in io.netty:netty@3.9.6.Final\n" +
-          "   introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.pinot:pinot-common@0.3.0 > io.netty:netty@3.9.6.Final")
+      because("Information Exposure [High Severity][https://snyk.io/vuln/SNYK-JAVA-IONETTY-30430] in io.netty:netty@3.9.6.Final")
     }
-
     implementation("org.glassfish.jersey.core:jersey-server:2.31") {
-      because("XML Entity Expansion [High Severity][https://snyk.io/vuln/SNYK-JAVA-ORGGLASSFISHJERSEYMEDIA-595972] in org.glassfish.jersey.media:jersey-media-jaxb@2.28\n" +
-              "    introduced by org.apache.pinot:pinot-tools@0.3.0 > org.apache.pinot:pinot-common@0.3.0 > org.glassfish.jersey.core:jersey-server@2.28 > org.glassfish.jersey.media:jersey-media-jaxb@2.28 and 1 other path(s)\n")
+      because("XML Entity Expansion [High Severity][https://snyk.io/vuln/SNYK-JAVA-ORGGLASSFISHJERSEYMEDIA-595972] in org.glassfish.jersey.media:jersey-media-jaxb@2.28")
     }
-    implementation("io.grpc:grpc-core@1.31.0") {
-      version {
-        because("Information Exposure [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-IOGRPC-571957] in io.grpc:grpc-core@1.30.0")
-        require("1.31.0")
-      }
+    implementation("io.grpc:grpc-netty-shaded:1.32.1") {
+      because("Information Exposure [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-IOGRPC-571957] in io.grpc:grpc-core@1.30.0")
     }
   }
```

Comment (tim-mwangi) [reply view-creator-framework/build.gradle.kts:98]: I liked the older way since we know the dependency path causing the constraint to be put in.

Comment (aaron-steinfeld) [reply view-creator-framework/build.gradle.kts:98]: we can add those back - the problem there is its often multiple paths (Snyk just gives the first), and those paths change as versions are changed

Comment (tim-mwangi) [reply view-creator-framework/build.gradle.kts:98]: yeah. the paths need to be updated. sigh.

File: view-creator-framework/src/test/java/org/hypertrace/core/viewcreator/pinot/PinotUtilsTest.java (hunk 1 above)
Comment (buchi-busireddy) [reply view-creator-framework/src/test/java/org/hypertrace/core/viewcreator/pinot/PinotUtilsTest.java:66]: yes, other places are good.

**Review by tim-mwangi — APPROVED** [review]

File: view-creator-framework/build.gradle.kts (hunk 3 above)
Comment (buchi-busireddy) [reply view-creator-framework/build.gradle.kts]: Thanks @aaron-steinfeld for the cleanup.
